image/daemon/image.go

```diff
@@ -92,8 +98,45 @@ func (img *image) ConfigFile() (*v1.ConfigFile, error) {
 		diffIDs = append(diffIDs, h)
 	}
 
-	// fill only RootFS in v1.ConfigFile
+	// fill only required metadata
+	var layersHistory []v1.History
+	var layerCreatedDate time.Time
+	var isEmptyLayer bool
+	for _, history := range img.history {
+		layerCreatedDate = time.Unix(history.Created, 0).UTC()
+		isEmptyLayer = false
+		// If layer size is zero it is empty layer.
+		if history.Size == 0 {
+			isEmptyLayer = true
+		}
+		configHistory := v1.History{
+			Created:    v1.Time{Time: layerCreatedDate},
+			CreatedBy:  history.CreatedBy,
+			Comment:    history.Comment,
+			EmptyLayer: isEmptyLayer,
+		}
+		layersHistory = append(layersHistory, configHistory)
+	}
+
+	// NOTE:  This is very, very important to understand time-parsing in go
+	loc, err := time.LoadLocation("")
+	if err != nil {
+		return nil, xerrors.Errorf("failed setting location %s: %w", img.inspect.Created, err)
+	}
+	time.Local = loc
+	imgCreatedDate, err := dateparse.ParseLocal(img.inspect.Created)
+	fmt.Println(imgCreatedDate)
+	if err != nil {
+		return nil, xerrors.Errorf("failed parsing created Date %s: %w", img.inspect.Created, err)
+	}
```

Comment:
Could you elaborate on it? Doesn't `time.ParseLocation(img.inspect.Created, time.UTC)` work?

Reply:
Sure, I assume you are asking about `ParseInLocation`  in https://golang.org/src/time/format.go.

It needs layout as input, that is the problem here. We don't know the date layout of created date.

Reply:
Why don't we know the date layout?